### PR TITLE
adds coverage reporting using nyc and coveralls.io

### PR DESCRIPTION
**Pull request by bcoe**
This pull adds coverage reporting, using coveralls.io and the [nyc](https://www.npmjs.com/package/nyc) instrumenter.

* run `npm run coverage` to get a human readable coverage report.
* run `npm run coverage -- --reporter=lcov` to get an HTML report over coverage in the /coverage folder.
* run `npm run coveralls` to report coverage to the [coveralls.io](https://coveralls.io/).
  * you'll need to setup your repo on coveralls.io and grap the `COVERALLS_REPO_TOKEN`.
  * on travis-ci.org, you'll need to set an environment variable with the value of `COVERALLS_REPO_TOKEN `

```shell
---------------|-----------|-----------|-----------|-----------|
File           |   % Stmts |% Branches |   % Funcs |   % Lines |
---------------|-----------|-----------|-----------|-----------|
   ./          |     98.08 |     77.78 |     90.91 |     98.08 |
      index.js |     98.08 |     77.78 |     90.91 |     98.08 |
---------------|-----------|-----------|-----------|-----------|
All files      |     98.08 |     77.78 |     90.91 |     98.08 |
---------------|-----------|-----------|-----------|-----------|
```

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks :)

File: readme.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 > Terminal string styling done right
 
 [![Build Status](https://travis-ci.org/sindresorhus/chalk.svg?branch=master)](https://travis-ci.org/sindresorhus/chalk) [![](http://img.shields.io/badge/unicorn-approved-ff69b4.svg)](https://www.youtube.com/watch?v=9auOCbH5Ns4)
+[![Coverage Status](https://coveralls.io/repos/sindresorhus/chalk/badge.svg?branch=)](https://coveralls.io/r/sindresorhus/chalk?branch=)
```
Comment (sindresorhus) [inline readme.md]: btw, the `branch` query param is empty.

Comment (bcoe) [inline readme.md]: I think I might have cargo culted this badge from an old version of the coveralls.io UI, which generated the badge in this way. I'll make sure to use `branch=master` as I instrument other codebases.

**Comment by bcoe** [comment]
thanks! \o/

**Comment by sindresorhus** [comment]
Would be happy to accept a similar PR on https://github.com/sindresorhus/pageres ;)

**Pull request merged**